### PR TITLE
Make istio-proxy's sub-server and prober can be uniform and compatible both for IPv4 and IPv6

### PR DESCRIPTION
**Pull request by zhlsunshine**
**Please provide a description of this PR:**

1. Make istio-proxy's sub-server can be compatible both for IPv4 and IPv6

>  status server
>  sts server
>  prober for health checking, readiness and liveness

2. Remove unnecessary wrapped IPv6 localhost

> Change wrapped IPv6 localhost from `[::1]` to `::1`,  this wrapped IPv6 localhost is specified as node.IPaddresses in some place , and then this IP address will be parsed by `net.ParseIP`, this will caused a IPv6 address can not be detected.

 

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [ ] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

**Comment by zhlsunshine** [comment]
Hi @istio/wg-security-maintainers & @istio/wg-networking-maintainers could you please help to check this PR when u r available? Thanks so much for your efforts! ^_^

File: pilot/cmd/pilot-agent/status/server_test.go [hunk 1]
```diff
@@ -666,7 +666,7 @@ func TestAppProbe(t *testing.T) {
 			name:       "tcp-livez-wrapped-ipv6",
 			probePath:  "app-health/hello-world/livez",
 			config:     simpleTCPConfig,
-			statusCode: http.StatusOK,
+			statusCode: http.StatusInternalServerError,
```
Comment (hzxuzhonghu) [inline pilot/cmd/pilot-agent/status/server_test.go:669]: why change it

Comment (hzxuzhonghu) [reply pilot/cmd/pilot-agent/status/server_test.go:669]: oh, [::1] is not a valid ip, so we can remove this case

**Review by hzxuzhonghu — APPROVED** [review]

**Review by howardjohn — APPROVED** [review]

**Pull request merged**